nuitka/importing/Importing.py

```diff
@@ -217,7 +217,13 @@ def findModule(importing, module_name, parent_package, level, warn):
     # We have many branches here, because there are a lot of cases to try.
     # pylint: disable=too-many-branches
 
-    assert type(module_name) is ModuleName
+    if type(module_name) is not ModuleName:
```

Comment:
That's definitely not the way to do it. You either obey to its demand, or you convert to `ModuleName` doing `module_name = ModuleName(module_name)` but you don't micro manage types like this.

I also think you are calling it with `ModuleName` in above code. Please just remove this change.

Reply:
Without this change, I get an `AssertionError`
```
Traceback (most recent call last):
  File "C:\Users\Tommy\Nuitka\tests\distutils\venv_nuitka\lib\site-packages\nuitka-0.6.6rc3-py3.7.egg\nuitka\__main__.py", line 184, in <module>
    main()
  File "C:\Users\Tommy\Nuitka\tests\distutils\venv_nuitka\lib\site-packages\nuitka-0.6.6rc3-py3.7.egg\nuitka\__main__.py", line 177, in main
    MainControl.main()
  File "C:\Users\Tommy\Nuitka\tests\distutils\venv_nuitka\lib\site-packages\nuitka-0.6.6rc3-py3.7.egg\nuitka\MainControl.py", line 753, in main
    main_module = createNodeTree(filename=filename)
  File "C:\Users\Tommy\Nuitka\tests\distutils\venv_nuitka\lib\site-packages\nuitka-0.6.6rc3-py3.7.egg\nuitka\MainControl.py", line 130, in createNodeTree
    warn=False,
  File "C:\Users\Tommy\Nuitka\tests\distutils\venv_nuitka\lib\site-packages\nuitka-0.6.6rc3-py3.7.egg\nuitka\importing\Importing.py", line 220, in findModule
    assert type(module_name) is ModuleName
AssertionError
```

I have no idea why, because as you said, it is called with `ModuleName` already...

Reply:
I added debug code to print out `type(module_name)` before the assertion. The result is `<class 'str'>`. Somehow `ModuleName` was not applied correct to the str? Im confused